app/src/shared/functions.ts

```diff
@@ -4,3 +4,13 @@ import { By } from '@angular/platform-browser';
 export const queryElement = (targetFixture: ComponentFixture<any>, targetClass: string) => {
   return targetFixture.debugElement.query(By.css(targetClass));
 };
+
+export const readFileAsString = (file: File): Promise<string> => {
```

Comment:
In general Angular prefers using Observables over Promises. https://angular.io/guide/comparing-observables
Maybe something to consider updating in a future cl.

Reply:
I can change it to an observable, my only obstacle is that these read methods have to wait to finish!